experiments/LES/dycoms.jl

```diff
@@ -0,0 +1,176 @@
+using Test
+using CLIMA
+using CLIMA.Atmos
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.MoistThermodynamics
+using CLIMA.PlanetParameters
+using CLIMA.VariableTemplates
+
+const ArrayType = CLIMA.get_array_type()
+
+"""
+  Initial Condition for DYCOMS_RF01 LES
+@article{doi:10.1175/MWR2930.1,
+author = {Stevens, Bjorn and Moeng, Chin-Hoh and Ackerman,
+          Andrew S. and Bretherton, Christopher S. and Chlond,
+          Andreas and de Roode, Stephan and Edwards, James and Golaz,
+          Jean-Christophe and Jiang, Hongli and Khairoutdinov,
+          Marat and Kirkpatrick, Michael P. and Lewellen, David C. and Lock, Adrian and
+          Maeller, Frank and Stevens, David E. and Whelan, Eoin and Zhu, Ping},
+title = {Evaluation of Large-Eddy Simulations via Observations of Nocturnal Marine Stratocumulus},
+journal = {Monthly Weather Review},
+volume = {133},
+number = {6},
+pages = {1443-1462},
+year = {2005},
+doi = {10.1175/MWR2930.1},
+URL = {https://doi.org/10.1175/MWR2930.1},
+eprint = {https://doi.org/10.1175/MWR2930.1}
+}
+"""
+function init_dycoms!(state::Vars, aux::Vars, (x,y,z), t)
+    FT            = eltype(state)
+
+    xvert::FT     = z
+    Rd::FT        = R_d
+
+    # These constants are those used by Stevens et al. (2005)
+    qref::FT      = FT(9.0e-3)
+    q_tot_sfc::FT = qref
+    q_pt_sfc      = PhasePartition(q_tot_sfc)
+    Rm_sfc::FT    = 461.5 #gas_constant_air(q_pt_sfc)
+    T_sfc::FT     = 292.5
+    P_sfc::FT     = 101780 #MSLP
+    ρ_sfc::FT     = P_sfc / Rm_sfc / T_sfc
+
+    # Specify moisture profiles
+    q_liq::FT      = 0
+    q_ice::FT      = 0
+    zb::FT         = 600         # initial cloud bottom
+    zi::FT         = 840         # initial cloud top
+    ziplus::FT     = 875
+    dz_cloud       = zi - zb
+    q_liq_peak::FT = 0.00045     # cloud mixing ratio at z_i
+    if xvert > zb && xvert <= zi
+        q_liq = (xvert - zb)*q_liq_peak/dz_cloud
+    end
+    if xvert <= zi
+        θ_liq = FT(289)
+        q_tot = qref
+    else
+        θ_liq = FT(297.5) + (xvert - zi)^(FT(1/3))
+        q_tot = FT(1.5e-3)
+    end
+
+    # Calculate PhasePartition object for vertical domain extent
+    q_pt  = PhasePartition(q_tot, q_liq, q_ice)
+    Rm    = gas_constant_air(q_pt)
+
+    # Pressure
+    H     = Rm_sfc * T_sfc / grav;
+    p     = P_sfc * exp(-xvert/H);
+    # Density, Temperature
+    T = air_temperature_from_liquid_ice_pottemp_given_pressure(θ_liq, p, q_pt)
+    ρ = air_density(T, p, q_pt)
+
+    # Assign State Variables
+    u1, u2 = FT(6), FT(7)
+    v1, v2 = FT(-4.25), FT(-5.5)
+    w = FT(0)
+    if (xvert <= zi)
+        u, v = u1, v1
+    elseif (xvert >= ziplus)
+        u, v = u2, v2
+    else
+        m = (ziplus - zi)/(u2 - u1)
+        u = (xvert - zi)/m + u1
+
+        m = (ziplus - zi)/(v2 - v1)
+        v = (xvert - zi)/m + v1
+    end
+    e_kin       = FT(1/2) * (u^2 + v^2 + w^2)
+    e_pot       = grav * xvert
+    E           = ρ * total_energy(e_kin, e_pot, T, q_pt)
```

Comment:
```suggestion
    E           = ρ * total_energy(e_kin, e_pot, ts)
```
Apply this if you were able to switch to using `ts = LiquidIcePotTempSHumEquil_given_pressure(θ_liq_ice, p, q_tot)`

Reply:
If the solution goes crazy, maybe try reverting this, since it's not precisely the same as what is in the [Dycoms driver](https://github.com/climate-machine/CLIMA/blob/sm/yt-DYCOMS-MULTI-RATE-NEW-MOIST-THERMO/test/DGmethods/compressible_Navier_Stokes/ex_001_dycoms-imex-FOR-MACIEJ.jl#L166)